tests/test_signal_processing.py

```diff
@@ -239,23 +239,68 @@ def test_gcc():
         assert lags[np.argmax(gccs)] == delay
 
 
-def test_all_tdoa_filter_types_find_correct_delay():
+def test_all_tdoa_filter_types_find_correct_delay_no_noise():
     delay = 20  # samples of delay (positive: second signal arrives before first)
     start = 500  # start of signal
     end = 510  # end of signal
 
     a = np.zeros(1000)
     a[start:end] = 3  # impulse
-    a += np.random.rand(1000)  # add noise
-    b = np.zeros(1000)
-    b[start - delay : end - delay] = 3
-    b += np.random.rand(1000)
+    reference_signal = np.zeros(1000)
+    reference_signal[start - delay : end - delay] = 3
+
+    # add max_delay samples to start and end of reference signal
+    max_delay = 50
+
+    for method in ["phat", "roth", "scot", "ht", "cc", "cc_norm"]:
+        estimated_sample_delay = sp.tdoa(
+            a,
+            reference_signal,
+            max_delay=max_delay,
+            cc_filter=method,
+            sample_rate=1,
+        )
+        assert estimated_sample_delay == delay
 
-    for method in ["phat", "roth", "scot", "ht", "cc"]:
-        assert sp.tdoa(a, b, cc_filter=method, sample_rate=1) == delay
 
-        # with sample rate !=1
-        assert sp.tdoa(a, b, cc_filter=method, sample_rate=22050) == delay / 22050
+def test_all_tdoa_filter_types_find_correct_delay_with_noise():
+    delay = 20  # samples of delay (positive: second signal arrives before first)
```

Comment:
"second signal arrives before first" is a mind-boggle. Can we use the terms `reference` and  `signal` (or `primary`) instead?

Reply:
I think it might be "+20 means the signal arrives 20 samples later than the reference".